### Measures and Central Tendency

In [76]:
# Step 1
salaries = [290000,245000,1200000,260000,270000,245000,245000,252000,220000]
print(salaries)

[290000, 245000, 1200000, 260000, 270000, 245000, 245000, 252000, 220000]


In [77]:
# Step 2
sum = 0
for salary in salaries:
    sum += salary
    mean = sum / len(salaries)
print("Mean = ", mean)
print("It does not look typical from my Step 1 eyeball check.")

Mean =  358555.55555555556
It does not look typical from my Step 1 eyeball check.


In [78]:
# Step 3
Sorted_list = sorted(salaries)
print("Sorted list:", Sorted_list)
print("There are 9 values which is odd therfore, the median is the single middle value. ")

middle_value = len(Sorted_list) // 2
median = Sorted_list[middle_value]
print("Median =", median)

Sorted list: [220000, 245000, 245000, 245000, 252000, 260000, 270000, 290000, 1200000]
There are 9 values which is odd therfore, the median is the single middle value. 
Median = 252000


In [79]:
# Step 4
counts = {}
for salary in salaries:
    if salary in counts:        
       counts[salary] += 1
    else:
       counts[salary] = 1
print(counts)

mode = None
highest_count = 0

for salary, count in counts.items():
    if count > highest_count:
        highest_count = count
        mode = salary
print()
print("Mode:", mode)

{290000: 1, 245000: 3, 1200000: 1, 260000: 1, 270000: 1, 252000: 1, 220000: 1}

Mode: 245000


In [80]:
# Step 5
print("Mean:", mean)
print("Median:", median)
print("Mode:", mode)

Mean: 358555.55555555556
Median: 252000
Mode: 245000


### Explanation
##### The median best represents a typical salary in this dataset.
##### The mean is misleading because every value contributes to the total, so the largest salary pulls the mean upward to about 358555.
##### The median is more resistant to the outlier because it depends only on the position of the values after they are sorted.
##### The mode is useful in this dataset because 245,000 occurs more often than any other salary.

In [81]:
# Step 7
new_list_salaries = [290000,245000,260000,270000,245000,245000,252000,220000]
print(new_list_salaries)

[290000, 245000, 260000, 270000, 245000, 245000, 252000, 220000]


In [82]:
   # Mean
sum = 0
for salary in new_list_salaries:
    sum += salary
    mean = sum / len(salaries)
print("Mean = ", mean)

   # Median
Sorted_list = sorted(new_list_salaries)
middle_value = len(Sorted_list) // 2
median = Sorted_list[middle_value]
print("Median =", median)

   # Mode
counts = {}
for salary in salaries:
    if salary in counts:        
       counts[salary] += 1
    else:
       counts[salary] = 1

mode = None
highest_count = 0
for salary, count in counts.items():
    if count > highest_count:
        highest_count = count
        mode = salary
print("Mode:", mode)

Mean =  225222.22222222222
Median = 252000
Mode: 245000


##### This test demonstrates that the mean is highly sensitive to extreme values. When the 1,200,000 outlier was removed, the mean dropped significantly from 358555 to 225222, while the median remained unchanged.